pyproject.toml

```diff
@@ -885,6 +885,7 @@ disable_error_code = [
     "has-type",
     "index",
     "list-item",
+    "method-assign",
```

Comment:
the code blocklist should not be added to -- weird that the script automatically did this :thinking:

Reply:
very strange -- will try and fix the script but in the meantime can you fix the `method-assign` that happened here?  if it's in a test it should use `with mock.patch.object(...):` to ensure that things are patched / unpatched correctly

Reply:
Addressed in https://github.com/getsentry/sentry/pull/56756/commits/83425bcd8dd2ebc68249b8bc782390b2b7efe361#diff-4ba1baf61a2a33dabab2a7843b5c5b4615dac4aa74fbfe9bc21e900b5683c215R107

In general, I sometimes use `contextlib.ExitStack` when some context manages may be conditional (rather than split the context tree, for instance).  In this case the patch is safe to retain in the `with` block; in the future, is using `mock.patch` conditionally in combination with an `ExitStack` not allowed?  Just curious so I can have the right mental model.

Reply:
I don't think that's the issue here -- I think it's the `try` / `finally` one -- `ExitStack` is fine and I [recommend it as well](https://youtu.be/FHixla7BLvI)

Reply:
https://github.com/getsentry/sentry/pull/56756/commits/7f5bda095e63317304c1fd3ad1b168a875fe0789